harvester/cli.py

```diff
@@ -57,17 +57,27 @@ def harvest(host, from_date, until, format, out, verbose):
                      "the arguments results in an empty list.")
         sys.exit()
 
+    identifier_list = []
+
+    for records in responses:
+        identifier_list.append(records.identifier)
+
+    logging.info(f"Identifier count to harvest: {len(identifier_list)}")
+
     with open(out, 'wb') as f:
         f.write('<records>'.encode())
 
-        for records in responses:
-            f.write(records.raw.encode('utf8'))
+        for identifier in identifier_list:
+            r = mysickle.GetRecord(identifier=identifier,
+                                   metadataPrefix=format)
+            f.write(r.raw.encode('utf8'))
             logging.debug(counter)
+            logging.debug(r.raw.encode('utf8'))
```

Comment:
You don't need to and really shouldn't encode the unicode here. The logger should be responsible for turning the unicode into bytes.